openreview/conference/templates/bidWebfield.js

```diff
@@ -21,55 +23,116 @@ function main() {
   load().then(renderContent).then(Webfield.ui.done);
 }
 
+function getPapersSortedByAffinity(offset) {
+  if (AFFINITY_SCORE_ID) {
+    return Webfield.get('/edges', {
+      invitation: AFFINITY_SCORE_ID,
+      tail: user.profile.id,
+      sort: 'weight:desc',
+      offset: offset,
+      limit: 50
+    })
+    .then(function(result) {
+      noteCount = result.count;
+      var edgesByHead = {};
+      var noteIds = result.edges.map(function(edge) {
+        edgesByHead[edge.head] = edge;
+        return edge.head;
+      });
+
+      return Webfield.post('/notes/search', {
+        ids: noteIds
+      })
+      .then(function(result) {
+        //Keep affinity score order
+        var mapById = result.notes.reduce(function(mapById, note) {
+          mapById[note.id] = note;
+          return mapById;
+        }, {});
+        return noteIds.map(function(id) {
+          return mapById[id];
+        });
+      })
+      .then(function(notes) {
```

Comment:
Since the previous `.then` doesn't perform any async actions is it possible to merge these 2 thens?

Reply:
you are right! thanks